sklearn/tests/test_kernel_approximation.py

```diff
@@ -114,7 +114,26 @@ def test_additive_chi2_sampler():
     Y_neg[0, 0] = -1
     msg = "Negative values in data passed to"
     with pytest.raises(ValueError, match=msg):
-        transform.transform(Y_neg)
+        transform.fit(X)
+
+    # test that the sample interval is set correctly
+    sample_steps_available = [1, 2, 3]
+    for sample_steps in sample_steps_available:
+
+        # test that the sample_interval is initialized correctly
+        transform = AdditiveChi2Sampler(sample_steps=sample_steps)
+        assert transform.sample_interval is None
```

Comment:
The correct assertion here is that `transform.sample_interval` should still be equal to the value passed in the constructor.

Reply:
Got it. I often wonder how much I should change things in the current logic.